juju/juju
# Use network SpaceInfo instead of interfaces where needed

This patch is an scaffolding preparation for the new dqlite domain for spaces, subnets and AZ.

The idea is to replace the usage of interfaces in favor of the core network.SpaceInfo in environs and apiserver facades. We need this because the new service and state in the domain will return core structs.

*Note: This patch follows https://github.com/juju/juju/pull/16615.*

## Checklist

*If an item is not applicable, use `~strikethrough~`.*


## QA steps

*Commands to run to verify that the change works.*

```
go test github.com/juju/juju/environs/space/... -gocheck.v
go test github.com/juju/juju/apiserver/facades/client/modelmanager/... -gocheck.v
```
Also, after bootstrapping on LXD:
Then add a space:
```
juju add-space test-space XXX.XXX.XXX.XXX/XX
# Make sure the space is listed:
juju spaces
```
Then remove it and make sure it's not listed any more:
```
juju remove-space test-space
juju spaces
```
only the `alpha` space should be present.

## Documentation changes


## Links


**Jira card:** JUJU-5054
